crypto/ecdsa/utils.go

```diff
@@ -0,0 +1,73 @@
+package ecdsa
+
+import (
+	"bufio"
+	"crypto/ecdsa"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/google/uuid"
+)
+
+func WriteKey(path string, privateKey *ecdsa.PrivateKey, password string) error {
+	UUID, err := uuid.NewRandom()
+	if err != nil {
+		return err
+	}
+	key := &keystore.Key{
+		Id:         UUID,
+		Address:    crypto.PubkeyToAddress(privateKey.PublicKey),
+		PrivateKey: privateKey,
+	}
+
+	encryptedBytes, err := keystore.EncryptKey(key, password, keystore.StandardScryptN, keystore.StandardScryptP)
+	if err != nil {
+		return err
+	}
+
+	return writeBytesToFile(path, encryptedBytes)
+}
+
+func writeBytesToFile(path string, data []byte) error {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0755); err != nil {
+		fmt.Println("Error creating directories:", err)
+		return err
+	}
+
+	file, err := os.Create(filepath.Clean(path))
+	if err != nil {
+		fmt.Println("file create error")
```

Comment:
this too

Reply:
this only created a file. This errors out if folder is not present which is why create before. This also errors out if it is not able to create file.

Reply:
Ah Create's doc actually notes that the file will be overwritten (truncated) https://pkg.go.dev/os#Create
Can we also add a comment to this function saying to be careful that if there is already a key stored at that path it will be overwritten?